src/qoi/src/rayfire_mesh.C

```diff
@@ -199,6 +198,35 @@ namespace GRINS
   }
 
 
+  const libMesh::Elem* RayfireMesh::get_start_elem(const libMesh::MeshBase& mesh_base)
+  {
+    const libMesh::Elem* start_elem = NULL;
+
+    libMesh::UniquePtr<libMesh::PointLocatorBase> locator = mesh_base.sub_point_locator();
+    const libMesh::Elem* elem = (*locator)(_origin);
+
+    if (this->rayfire_in_elem(_origin,elem))
+      start_elem = elem;
+    else
+      {
+        for (unsigned int i=0; i<elem->n_neighbors(); i++)
+          {
+            const libMesh::Elem* neighbor_elem = elem->neighbor(i);
+            if (!neighbor_elem)
+              continue;
+
+            if (this->rayfire_in_elem(_origin,neighbor_elem))
+              {
+                start_elem = neighbor_elem;
+                break;
+              }
+          }
+      }
+
```

Comment:
I check the return value of `get_start_elem()` in `init()` line 85, because it can return `NULL` if the origin is not on the mesh, in which case, I `libmesh_error_msg`